certbot-apache/certbot_apache/configurator.py

```diff
@@ -886,70 +978,85 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
         """
         # First register the creation so that it is properly removed if
         # configuration is rolled back
-        self.reverter.register_file_creation(False, ssl_fp)
+        if os.path.exists(ssl_fp):
+            notes = "Appended new VirtualHost directive to file %s" % ssl_fp
+            files = set()
+            files.add(ssl_fp)
+            self.reverter.add_to_checkpoint(files, notes)
+        else:
+            self.reverter.register_file_creation(False, ssl_fp)
         sift = False
 
         try:
             with open(avail_fp, "r") as orig_file:
-                with open(ssl_fp, "w") as new_file:
-                    new_file.write("<IfModule mod_ssl.c>\n")
-
-                    comment = ("# Some rewrite rules in this file were "
-                              "disabled on your HTTPS site,\n"
-                              "# because they have the potential to create "
-                              "redirection loops.\n")
-
-                    for line in orig_file:
-                        A = line.lstrip().startswith("RewriteCond")
-                        B = line.lstrip().startswith("RewriteRule")
-
-                        if not (A or B):
-                            new_file.write(line)
-                            continue
+                orig_file_list = [line for line in orig_file]
+                if vhost_num != -1:
+                    orig_file_list = self._create_block_segments(orig_file_list, vhost_num)
```

Comment:
I can't believe I missed this! You are absolutely right. I played around with some ways of getting the correct vhost, and this suggestion seems to be the best way. However I would like to implement it in to VirtualHost object creation so that every VirtualHost object would read in, and carry their own configuration block through the execution. This will affect the _copy_create_ssl_vhost_skeleton rewrite as well, and will most likely simplify the flow when writing out the new configuration.